backend/src/Designer/Infrastructure/GitRepository/AltinnAppGitRepository.cs

```diff
@@ -804,19 +820,66 @@ public Stream GetProcessDefinitionFile()
             return OpenStreamByRelativePath(ProcessDefinitionFilePath);
         }
 
+        /// <summary>
+        /// Checks if image already exists in wwwroot
+        /// </summary>
+        /// <param name="imageFilePath">The file path of the image from wwwroot</param>
+        /// <returns>A boolean indication if image exists</returns>
+        public bool DoesImageExist(string imageFilePath)
+        {
+            return FileExistsByRelativePath(GetPathToImage(imageFilePath));
+        }
+
         /// <summary>
         /// Gets specified image from App/wwwroot folder of local repo
         /// </summary>
         /// <param name="imageFilePath">The file path of the image</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> that observes if operation is cancelled.</param>
         /// <returns>The image as stream</returns>
-        public Stream GetImage(string imageFilePath, CancellationToken cancellationToken = default)
+        public Stream GetImageAsStreamByFilePath(string imageFilePath, CancellationToken cancellationToken = default)
         {
             cancellationToken.ThrowIfCancellationRequested();
             string imagePath = GetPathToImage(imageFilePath);
             return OpenStreamByRelativePath(imagePath);
         }
 
+        /// <summary>
+        /// Delete specified image from App/wwwroot folder of local repo
+        /// </summary>
+        /// <param name="imageFilePath">The file path of the image</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that observes if operation is cancelled.</param>
+        /// <returns>The image as stream</returns>
+        public Task DeleteImageByImageFilePath(string imageFilePath, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            string imagePath = GetPathToImage(imageFilePath);
+            DeleteFileByRelativePath(imagePath);
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Gets all image filePathNames from App/wwwroot folder of local repo
+        /// </summary>
+        /// <returns>Array of file paths to all images in App/wwwroot</returns>
+        public List<string> GetAllImageFileNames()
+        {
+            List<string> allFilePaths = new List<string>();
+
+            if (!DirectoryExistsByRelativePath(ImagesFolderName))
+            {
+                return allFilePaths;
+            }
+
+            string[] allowedExtensions = { ".png", ".jpg", ".jpeg", ".svg", ".gif", ".bmp", ".webp", ".tiff" };
```

Comment:
One issue I see with this list is that it is limited to some extensions, while on the frontend it's possible to upload any type of image. This means that if I upload an icon (e.g., favicon.ico), it won't appear in the dialog

Reply:
Agree, but it did not seem possible to use `image/*` in backend 🤔